### PR TITLE
Fix 7 - Export MFF into JSON format

### PR DESCRIPTION
**Pull request by damian5710**
This pull request addresses Issue #7 .

It adds the ability to export the contents of an MFF file into a JSON format.

Please, note that the exported data will be the content of those .xml files which can currently be read by this module.

Note: for now, the JSON exporting feature only works for segmented mffs files.

### CODE REVIEW

File: bin/mff2json.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@author: Damian Persico - persicodamian@gmail.com
+"""
+
+from os.path import splitext
+from mffpy import Reader, Writer
+from argparse import ArgumentParser
+
+
+def parse_args():
+    parser = ArgumentParser()
```
Comment (dhorkin) [inline bin/mff2json.py]: unless we're planning on having more arguments in the future, since this is only used the one time and effectively is just two lines of code, I'd remove this function. Additionally, I'd move line 36-52. End would look like this:

```Python3.6
def mff2json(input_filename):
    # lines 36-52 go here

if __name__ == "__main__":
    parser = ArgumentParser()
    parser.add_argument("--mff_file", type=str, help="Path to an MFF directory")
    mff2json(parser.mff_file)
```

File: bin/mff2json.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    parser.add_argument("--mff_file", type=str, help="Path to an MFF directory")
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    input_filename = parse_args().mff_file
+
+    if input_filename:
+        # Check .mff input
+        dir_and_base, ext = splitext(input_filename.rstrip('/'))
```
Comment (dhorkin) [inline bin/mff2json.py]: Use `input_filename.rstrip(os.path.sep)` instead of `input_filename.rstrip('/')` so that it's platform independent

File: mffpy/xml_files.py [hunk 3]
```diff
@@ -289,6 +309,25 @@ def content(cls, fileDataType: str,  # type: ignore
...
+
+    def get_serializable_content(self):
+        """return a serializable object containing
+        info on the associated (data) .bin file"""
+        content = copy.deepcopy(self.get_content())
+        # Convert np.float32 values to float built-in type
+        for value in content['calibrations'].values():
+            channels = value['channels']
+            for channel in channels.keys():
+                channels[channel] = float(channels[channel])
```
Comment (dhorkin) [inline mffpy/xml_files.py]: Why is this necessary? I made a toy example and it looks like np.float32 and floats are generally directly comparable. Do we need 64-bit precision or something? (looks like that's _generally_ the default for floats in python these days https://docs.scipy.org/doc/numpy/user/basics.types.html suggests we need to do this )

File: mffpy/reader.py [hunk 4]
```diff
@@ -236,3 +236,54 @@ def get_physical_samples_from_epoch(self, epoch: xml_files.Epoch,
...
+                        # Add EEG data to each segment of each category
+                        for cat_obj in content['categories'].values():
+                            # Iterate over each segment
+                            for i in range(len(cat_obj)):
+                                # Get samples from epoch
+                                samples = self.get_physical_samples_from_epoch(
+                                    self.epochs[i], channels=['EEG'])
+                                eeg, start_time = samples['EEG']
+                                # Insert an EEG data field into each segment
+                                cat_obj[i]['eegData'] = eeg.tolist()
```
Comment (dhorkin) [inline mffpy/reader.py]: Any reason this isn't in `Categories.get_content` or `Categories.get_serializable_content`?

File: requirements.txt [hunk 5]
```diff
@@ -1,6 +1,6 @@
 mypy==0.660
 pytz==2019.2
 numpy==1.15.1
-pytest==4.1.1
+pytest==5.2.0
```
[the remark this answers is not in the document]
Comment (damian5710) [reply requirements.txt]: Yes, I upgraded pytest version number because the previous one was giving me an error when running the tests (on Windows 10):

```
$ mingw32-make test
mypy --ignore-missing-imports mffpy
pytest --cov
Traceback (most recent call last):
  File "c:\users\damian\anaconda3\envs\mffpy\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\damian\anaconda3\envs\mffpy\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Damian\Anaconda3\envs\mffpy\Scripts\pytest.exe\__main__.py", line 7, in <module>
  File "c:\users\damian\anaconda3\envs\mffpy\lib\site-packages\_pytest\config\__init__.py", line 61, in main
    config = _prepareconfig(args, plugins)
  File "c:\users\damian\anaconda3\envs\mffpy\lib\site-packages\_pytest\config\__init__.py", line 182, in _prepareconfig
    config = get_config()
  File "c:\users\damian\anaconda3\envs\mffpy\lib\site-packages\_pytest\config\__init__.py", line 156, in get_config
    pluginmanager.import_plugin(spec)
  File "c:\users\damian\anaconda3\envs\mffpy\lib\site-packages\_pytest\config\__init__.py", line 530, in import_plugin
    __import__(importspec)
  File "c:\users\damian\anaconda3\envs\mffpy\lib\site-packages\_pytest\tmpdir.py", line 25, in <module>
    class TempPathFactory(object):
  File "c:\users\damian\anaconda3\envs\mffpy\lib\site-packages\_pytest\tmpdir.py", line 35, in TempPathFactory
    lambda p: Path(os.path.abspath(six.text_type(p)))
TypeError: attrib() got an unexpected keyword argument 'convert'
makefile:10: recipe for target 'test' failed
mingw32-make: *** [test] Error 1
```

Then I did a quick search and I found [this](https://stackoverflow.com/a/58189684) answer suggesting that the best action is to update pytest to a more recent version.

All tests are passing on Windows 10 now.

File: bin/mff2json.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        # Write data into a JSON file
+        writer = Writer(output_filename)
+        print("Writing data out to " + str(output_filename))
+        writer.export_to_json(data)
+    else:
+        print("No .mff file provided.\nPlease, provide a path to a valid .mff directory")
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument("--mff_file", type=str, help="Path to an MFF directory")
```
[the remark this answers is not in the document]
Comment (damian5710) [reply bin/mff2json.py]: Yes, I agree. Thanks!

File: setup.py [hunk 7]
```diff
@@ -4,7 +4,7 @@
     name='mffpy',
     version='0.3.0',
```
[the remark this answers is not in the document]
Comment (damian5710) [reply setup.py]: Done!

File: requirements.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (ViridianForge) [reply requirements.txt]: Sounds good man - we can consider this documented, but in the future when we update package requirements in `Requirements.txt`,  we should be in the habit of documenting why the change was made in the relevant commit.

[the remark this answers is not in the document]
Comment (damian5710) [reply requirements.txt]: Will do!
